vms/proposervm/proposer/windower.go

```diff
@@ -51,6 +62,23 @@ type Windower interface {
 		validatorID ids.NodeID,
 		maxWindows int,
 	) (time.Duration, error)
+
+	ExpectedProposer(
+		ctx context.Context,
+		chainHeight,
+		pChainHeight uint64,
+		parentBlockTime,
+		blockTime time.Time,
+	) (ids.NodeID, error)
+
+	MinDelayForProposer(
+		ctx context.Context,
+		chainHeight,
+		pChainHeight uint64,
+		parentBlockTime time.Time,
+		nodeID ids.NodeID,
+		startTime time.Time,
+	) (time.Time, error)
```

Comment:
It feels a bit odd that `Delay` returns a `time.Duration` but this returns a `time.Time`... I think it makes sense... But just a little jarring

Reply:
Reworked to return a delay. Client will have to add to parentBlockTime.